charts/home-assistant/templates/servicemonitor.yaml

```diff
@@ -1,43 +1,32 @@
-{{- if and ( .Values.monitoring.serviceMonitor.enabled ) ( .Values.monitoring.enabled ) }}
+{{- if .Values.prometheus.serviceMonitor.enabled }}
 apiVersion: monitoring.coreos.com/v1
 kind: ServiceMonitor
 metadata:
-{{- if .Values.monitoring.serviceMonitor.labels }}
+  name: {{ template "common.names.fullname" . }}
   labels:
-{{ toYaml .Values.monitoring.serviceMonitor.labels | indent 4}}
-{{- end }}
-  name: {{ template "home-assistant.fullname" . }}-prometheus-exporter
-{{- if .Values.monitoring.serviceMonitor.namespace }}
-  namespace: {{ .Values.monitoring.serviceMonitor.namespace }}
-{{- end }}
+    {{- include "common.labels" . | nindent 4 }}
+    {{- with .Values.prometheus.serviceMonitor.additionalLabels }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
 spec:
+  selector:
+    matchLabels:
+      {{- include "common.labels.selectorLabels" . | nindent 6 }}
   endpoints:
-  - targetPort: {{ .Values.service.portName }}
-    path: /api/prometheus
-{{- if .Values.monitoring.serviceMonitor.interval }}
-    interval: {{ .Values.monitoring.serviceMonitor.interval }}
-{{- end }}
-{{- if .Values.monitoring.serviceMonitor.bearerTokenFile }}
-    bearerTokenFile: {{ .Values.monitoring.serviceMonitor.bearerTokenFile }}
-{{- end }}
-{{- if .Values.monitoring.serviceMonitor.bearerTokenSecret }}
+  - port: http
```

Comment:
it may be better to refer to `{{ default "http" .Values.service.port.name }}`

Reply:
Agreed!